File: src/main/java/withoutaname/mods/immersivesignals/setup/ModSetup.java
package withoutaname.mods.immersivesignals.setup;

import javax.annotation.Nonnull;

import net.minecraft.world.item.CreativeModeTab;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;
import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;

import withoutaname.mods.immersivesignals.modules.signal.SignalRegistration;
import withoutaname.mods.immersivesignals.modules.signalcontroller.network.SignalControllerNetworking;

public class ModSetup {
	
	public static final CreativeModeTab DEFAULT_CREATIVE_TAB = new CreativeModeTab("Immersive Signals") {
		
		@Nonnull
		@Override
		public ItemStack makeIcon() {
			return new ItemStack(SignalRegistration.SIGNAL_ITEM.get());
		}
		
	};
	
	public static void init(FMLCommonSetupEvent event) {
		SignalControllerNetworking.registerMessages();
	}
	
	public static final Item.Properties DEFAULT_ITEM_PROPERTIES = new Item.Properties().tab(DEFAULT_CREATIVE_TAB);
	
}

File: src/main/kotlin/eu/withoutaname/mod/immersivesignals/setup/ModSetup.kt
package eu.withoutaname.mod.immersivesignals.setup

import eu.withoutaname.mod.immersivesignals.ImmersiveSignals
import net.minecraft.item.Item
import net.minecraft.item.ItemGroup
import net.minecraft.item.ItemStack

object ModSetup {

    val DEFAULT_CREATIVE_TAB: ItemGroup = object : ItemGroup(ImmersiveSignals.ID) {
        override fun makeIcon(): ItemStack {
            return ItemStack(Registration.signalItem)
        }
    }
    val DEFAULT_ITEM_PROPERTIES: Item.Properties = Item.Properties().tab(DEFAULT_CREATIVE_TAB)
}
